src/datachain/query/batch.py

```diff
@@ -59,31 +67,24 @@ def __init__(self, count: int):
 
     def __call__(
         self,
-        execute: Callable,
-        query: sa.sql.selectable.Select,
-    ) -> Generator[RowBatch, None, None]:
+        execute: Callable[..., Generator[Sequence, None, None]],
+        query: "Select",
+    ) -> Generator[RowsInputBatch, None, None]:
         # choose page size that is a multiple of the batch size
         page_size = math.ceil(SELECT_BATCH_SIZE / self.count) * self.count
 
         # select rows in batches
-        results: list[RowDict] = []
-
-        with contextlib.closing(
-            execute(
-                query,
-                page_size=page_size,
-                limit=query._limit,
```

Comment:
why is it safe / where do we now do limit / order by?

Reply:
`execute` here is `dataset_select_paginated`, and it is related to this change here: https://github.com/iterative/datachain/pull/211/files/082757b233a2621a467c29f11a2181cd466ab6b8#diff-18ccbe3cf14ab41f123a6c5ec1848c31ddb1549e578ce4ce9db82622f09d994eL200-L206

Previously we run `dataset_select_paginated` with `limit` and `order_by` taken from `query` (it was a reason before, but now it is not). Now we are passing `query` only and take `limit` and `order_by` from query inside `dataset_select_paginated`.